internal/bridge/managedpostgrescluster/managedpostgrescluster_controller.go

```diff
@@ -0,0 +1,552 @@
+				// Requeue now that we have a cluster ID assigned
+				return ctrl.Result{Requeue: true}, nil
+			}
+		}
+
+		// END FIND
+
+		// if we've gotten here then no cluster exists with that name and we're missing the ID, ergo, create cluster
+
+		// TODO(managedpostgrescluster): gotta investigate this: storage always in Gi?
+		// We probably want to emphasize consistency between our APIs, e.g., storage `10Gi` not `10`
+		// When do we deviate from our API or Bridge API?
+		val, _ := managedpostgrescluster.Spec.Storage.AsInt64()
+
+		// TODO(managedpostgrescluster) Can almost just use the managed.Spec... except for the team, which we don't want
+		// users to set on the spec. Do we?
+		clusterReq := &v1beta1.ClusterDetails{
+			IsHA:            managedpostgrescluster.Spec.IsHA,
+			Name:            managedpostgrescluster.Spec.ClusterName,
+			Plan:            managedpostgrescluster.Spec.Plan,
+			PostgresVersion: intstr.FromInt(managedpostgrescluster.Spec.PostgresVersion),
+			Provider:        managedpostgrescluster.Spec.Provider,
+			Region:          managedpostgrescluster.Spec.Region,
+			Storage:         int(val),
+			Team:            team,
+		}
+		cluster, err := r.NewClient().CreateCluster(ctx, key, clusterReq)
+		if err != nil {
+			log.Error(err, "whoops, cluster creating issue")
+			// TODO(managedpostgrescluster): probably shouldn't set this condition unless response from Bridge
+			// indicates the payload is wrong
+			// Otherwise want a different condition
+			meta.SetStatusCondition(&managedpostgrescluster.Status.Conditions, metav1.Condition{
+				ObservedGeneration: managedpostgrescluster.GetGeneration(),
+				Type:               v1beta1.ConditionCreating,
+				Status:             metav1.ConditionFalse,
+				Reason:             "ClusterInvalid",
+				Message: fmt.Sprintf(
+					"Cannot create from spec for some reason: %v", "TODO(managedpostgrescluster)"),
+			})
+
+			// TODO(managedpostgrescluster): If the payload is wrong, we don't want to requeue, so pass nil error
+			// If the transmission hit a transient problem, we do want to requeue
+			return ctrl.Result{}, nil
+		}
+		managedpostgrescluster.Status.ID = cluster.ID
+		return ctrl.Result{RequeueAfter: 3 * time.Minute}, nil
+	}
+
+	// If we reach this point, our ManagedPostgresCluster object has an ID
+	// so we want to fill in the details for the cluster and cluster upgrades from the Bridge API
+	// Consider cluster details as a separate func.
+	clusterDetails, err := r.NewClient().GetCluster(ctx, key, managedpostgrescluster.Status.ID)
+	if err != nil {
+		log.Error(err, "whoops, cluster getting issue")
+		return ctrl.Result{}, err
+	}
+	managedpostgrescluster.Status.Cluster = clusterDetails
+
+	clusterUpgradeDetails, err := r.NewClient().GetClusterUpgrade(ctx, key, managedpostgrescluster.Status.ID)
+	if err != nil {
+		log.Error(err, "whoops, cluster upgrade getting issue")
+		return ctrl.Result{}, err
+	}
+	managedpostgrescluster.Status.ClusterUpgrade = clusterUpgradeDetails
```

Comment:
Do we want to keep the history of upgrades somewhere? On the status in a different field, or as events?

Reply:
Bridge Audit log:

<img width="1429" alt="Screenshot 2024-01-30 at 2 41 43 PM" src="https://github.com/CrunchyData/postgres-operator/assets/4651855/99ff8575-e426-41ad-885e-25592e757700">

Reply:
Created a ticket